x-pack/plugins/security_solution/public/flyout/document_details/right/components/highlighted_fields_cell.tsx

```diff
@@ -84,34 +87,54 @@ export const HighlightedFieldsCell: VFC<HighlightedFieldsCellProps> = ({
   values,
   field,
   originalField,
-}) => (
-  <>
-    {values != null &&
-      values.map((value, i) => {
-        return (
-          <EuiFlexItem
-            grow={false}
-            key={`${i}-${value}`}
-            data-test-subj={`${value}-${HIGHLIGHTED_FIELDS_CELL_TEST_ID}`}
-          >
-            {field === HOST_NAME_FIELD_NAME || field === USER_NAME_FIELD_NAME ? (
-              <LinkFieldCell value={value} />
-            ) : field === AGENT_STATUS_FIELD_NAME &&
-              originalField === SENTINEL_ONE_AGENT_ID_FIELD ? (
-              <SentinelOneAgentStatus
-                agentId={String(value ?? '')}
-                data-test-subj={HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID}
-              />
-            ) : field === AGENT_STATUS_FIELD_NAME ? (
-              <EndpointAgentStatusById
-                endpointAgentId={String(value ?? '')}
-                data-test-subj={HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID}
-              />
-            ) : (
-              <span data-test-subj={HIGHLIGHTED_FIELDS_BASIC_CELL_TEST_ID}>{value}</span>
-            )}
-          </EuiFlexItem>
-        );
-      })}
-  </>
-);
+}) => {
+  const agentStatusClientEnabled = useIsExperimentalFeatureEnabled('agentStatusClientEnabled');
+
+  const getAgentStatus = useCallback(
+    (value?: string) => {
+      if (field === AGENT_STATUS_FIELD_NAME) {
+        const isSentinelOneAgentIdField = originalField === SENTINEL_ONE_AGENT_ID_FIELD;
+        if (isSentinelOneAgentIdField || agentStatusClientEnabled) {
+          return (
+            <AgentStatus
+              agentId={String(value ?? '')}
+              agentType={isSentinelOneAgentIdField ? 'sentinel_one' : 'endpoint'}
+              data-test-subj={HIGHLIGHTED_FIELDS_AGENT_STATUS_CELL_TEST_ID}
+            />
+          );
+        } else {
+          // TODO: remove usage of `EndpointAgentStatusById` when `agentStatusClientEnabled` FF is enabled
```

Comment:
I would suggest to remove usage of `EndpointAgentStatusById` when the `agentStatusClientEnabled` FF gets **removed**, not **enabled**.